hwcomposer/wayland-hwc.cpp

```diff
@@ -772,12 +738,11 @@ keyboard_handle_enter(void *data, struct wl_keyboard *,
     display->keyboard_enter_serial = serial;
 
     std::scoped_lock lock(display->windowsMutex);
-    if (display->windows.find(surface) == display->windows.end())
-        return;
-
-    struct window *window = display->windows[surface];
+    auto window = reinterpret_cast<struct window *>(wl_surface_get_user_data(surface));
+    if (!window)
+        return; // Window was destroyed in hwc_set
 
-    if (window->display->task != nullptr) {
+    if (display->task != nullptr) {
         if (window->taskID != "none" && window->taskID != "0") {
```

Comment:
s/none/None/ to match what you're using in `split_layer_names_helper`
Actually this should probably be made resistant to non-integers and avoid stoi exceptions

Reply:
Ups. This change was unintentional. Changed it back to "none" in `split_layer_names_helper`